core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -140,8 +140,70 @@ export class AdminBackendApiService {
       });
     });
   }
-}
+  flushCache():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_MEMORY_CACHE_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  clearSearchIndex():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  regenerateTopicRelatedOpportunities(topicId:string):Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {
+        action: 'regenerate_topic_related_opportunities',
+        topic_id: topicId
+      }
+    ).toPromise();
+  }
+
+  uploadTopicSimilarities(data:string | ArrayBuffer):Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_HANDLER_URL, {
+        action: 'upload_topic_similarities',
+        data: data,
+      }
+    ).toPromise();
+  }
 
+  sendDummyMail():Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_SEND_DUMMY_MAIL_HANDLER_URL, {}
+    ).toPromise();
+  }
+
+  getMemoryCacheProfile():Promise<void> {
+    let memoryCacheProfile = null;
+    return new Promise((resolve, reject) => {
+      this.http.get<void>(
+        AdminPageConstants.ADMIN_MEMORY_CACHE_HANDLER_URL, {}
+      ).toPromise()
+        .then(responce =>{
+          memoryCacheProfile = responce;
+          resolve(memoryCacheProfile);
+        }, errorResponse => reject(errorResponse));
+    }
+    );
+  }
+  updateUserName(oldUsername:string, newUsername:string):Promise<void> {
+    return this.http.put<void>(
+      AdminPageConstants.ADMIN_UPDATE_USERNAME_HANDLER_URL, {
+        old_username: oldUsername,
+        new_username: newUsername
+      }
+    ).toPromise();
+  }
+
+  numberOfPendingDeletionRequest():Promise<void> {
+    return this.http.get<void>(
```

Comment:
This is a get request so I'm expecting to receive a dict response here, right?

Reply:
done!

Reply:
![pendingRequests](https://user-images.githubusercontent.com/54100702/103095349-52fdd300-4626-11eb-8304-f97d60ad816d.png)

Reply:
Define an interface for this dict and use it here instead of calling the type to be Object.